misc/scripts/purge_cache.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import os
+
+if __name__ != "__main__":
+    raise ImportError(f"{__name__} should not be used as a module.")
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Cleanup old cache files")
+    parser.add_argument("timestamp", type=int, help="Unix timestamp cutoff")
+    parser.add_argument("directory", help="Path to cache directory")
+    args = parser.parse_args()
+
+    ret = 0
+
+    for file in glob.glob(os.path.join(args.directory, "*", "*")):
+        try:
+            if os.path.getatime(file) < args.timestamp:
```

Comment:
My concern here is that this will include:
* Files from `master`
* Files created in the *last* run
* Files created in the *current* run

But not:
* Files created in some other previous run

So if we push a PR twice it will not keep cache files from the first run that was then reused in the second, in some cases this could be pretty major, for example a PR that, in the first push changes `object.h` and then in a second just changes a single line in some random file, and then makes another minor change, in this case only the `master` state of the cache and the first small change would be preserved